diff --git a/node_modules/@capacitor/android/capacitor/src/main/java/com/getcapacitor/CapacitorWebView.java b/node_modules/@capacitor/android/capacitor/src/main/java/com/getcapacitor/CapacitorWebView.java
index 8f8a8f8..a1b2c3d 100644
--- a/node_modules/@capacitor/android/capacitor/src/main/java/com/getcapacitor/CapacitorWebView.java
+++ b/node_modules/@capacitor/android/capacitor/src/main/java/com/getcapacitor/CapacitorWebView.java
@@ -59,15 +59,7 @@ public class CapacitorWebView extends WebView {
 
         if (configEdgeToEdge.equals("disable")) return;
 
-        boolean autoMargins = false;
-        boolean forceMargins = configEdgeToEdge.equals("force");
-
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM && configEdgeToEdge.equals("auto")) {
-            TypedValue value = new TypedValue();
-            boolean foundOptOut = getContext().getTheme().resolveAttribute(android.R.attr.windowOptOutEdgeToEdgeEnforcement, value, true);
-            boolean optOutValue = value.data != 0; // value is set to -1 on true as of Android 15, so we have to do this.
-
-            autoMargins = !(foundOptOut && optOutValue);
-        }
+        boolean forceMargins = configEdgeToEdge.equals("force") || configEdgeToEdge.equals("auto");
 
         if (forceMargins || autoMargins) {
             ViewCompat.setOnApplyWindowInsetsListener(this, (v, windowInsets) -> {
